File: Source/Addins/MonoDevelop.GtkSharp.Addin/Templates/File/Data/Dialog.cs
using System;
using Gtk;
using UI = Gtk.Builder.ObjectAttribute;

namespace ${Namespace}
{
    class ${EscapedIdentifier} : Dialog
    {
        public MyDialog() : this(new Builder("${Namespace}.${EscapedIdentifier}.glade")) { }

        private MyDialog(Builder builder) : base(builder.GetObject("${EscapedIdentifier}").Handle)
        {
            builder.Autoconnect(this);
            DefaultResponse = ResponseType.Cancel;

            Response += OnResponse;
        }

        private void OnResponse(object o, ResponseArgs args)
        {
            Hide();
        }
    }
}

File: Source/Addins/MonoDevelop.GtkSharp.Addin/Templates/File/Data/Dialog.fs
namespace ${Namespace}

open Gtk

type ${EscapedIdentifier} (builder : Builder) as this =
    inherit Dialog(builder.GetObject("${EscapedIdentifier}").Handle)
    do
        this.DefaultResponse <- ResponseType.Cancel;
        this.Response.Add(fun _ ->
            this.Hide();
        )

    new() = new ${EscapedIdentifier}(new Builder("${Namespace}.${EscapedIdentifier}.glade"))
